src/main/java/org/jabref/logic/citation/repository/MVStoreBibEntryRelationDAO.java

```diff
@@ -0,0 +1,238 @@
+                entry.getField(StandardField.URL).orElse("null"),
+                entry.getField(StandardField.ABSTRACT).orElse("null")
+            );
+        }
+
+        private static Optional<String> extractFieldValue(String field) {
+            return Objects.equals(field, "null") || field == null
+                ? Optional.empty()
+                : Optional.of(field);
+        }
+
+        private static BibEntry fromString(String serializedString) {
+            var fields = serializedString.split(FIELD_SEPARATOR);
+            BibEntry entry = new BibEntry();
+            extractFieldValue(fields[0]).ifPresent(title -> entry.setField(StandardField.TITLE, title));
+            extractFieldValue(fields[1]).ifPresent(year -> entry.setField(StandardField.YEAR, year));
+            extractFieldValue(fields[2]).ifPresent(authors -> entry.setField(StandardField.AUTHOR, authors));
+            extractFieldValue(fields[3]).ifPresent(type -> entry.setType(StandardEntryType.valueOf(type)));
+            extractFieldValue(fields[4]).ifPresent(doi -> entry.setField(StandardField.DOI, doi));
+            extractFieldValue(fields[5]).ifPresent(url -> entry.setField(StandardField.URL, url));
+            extractFieldValue(fields[6])
+                .ifPresent(entryAbstract -> entry.setField(StandardField.ABSTRACT, entryAbstract));
+            return entry;
+        }
+
+        @Override
+        public int getMemory(BibEntry obj) {
+            return toString(obj).getBytes(StandardCharsets.UTF_8).length;
+        }
+
+        @Override
+        public void write(WriteBuffer buff, BibEntry bibEntry) {
+            var asBytes = toString(bibEntry).getBytes(StandardCharsets.UTF_8);
+            buff.putInt(asBytes.length);
+            buff.put(asBytes);
+        }
+
+        @Override
+        public BibEntry read(ByteBuffer buff) {
+            int serializedEntrySize = buff.getInt();
+            var serializedEntry = new byte[serializedEntrySize];
+            buff.get(serializedEntry);
+            return fromString(new String(serializedEntry, StandardCharsets.UTF_8));
+        }
+
+        @Override
+        public int compare(BibEntry a, BibEntry b) {
+            if (a == null || b == null) {
+                throw new NullPointerException();
+            }
+            return toString(a).compareTo(toString(b));
+        }
+
+        @Override
+        public BibEntry[] createStorage(int size) {
+            return new BibEntry[size];
+        }
+
+        @Override
+        public boolean isMemoryEstimationAllowed() {
+            return false;
+        }
+    }
+
+    private static class BibEntryHashSetSerializer extends BasicDataType<LinkedHashSet<BibEntry>> {
+
+        private final BasicDataType<BibEntry> bibEntryDataType = new BibEntrySerializer();
+
+        /**
+         * Memory size is the sum of all aggregated bibEntries memory size plus 4 bytes.
+         * Those 4 bytes are used to store the length of the collection itself.
```

Comment:
It's better to move this explanation inside this `getMemory()` function as it contains only implementation details